src/framework/theme/components/select/option-group.component.spec.ts

```diff
@@ -0,0 +1,148 @@
+
+        <nb-select [disabled]="selectDisabled">
+          <nb-option-group [disabled]="optionGroupDisabled" [title]="optionGroupTitle">
+            <nb-option *ngIf="showOption" [value]="1" [disabled]="optionDisabled">1</nb-option>
+          </nb-option-group>
+        </nb-select>
+
+      </nb-layout-column>
+    </nb-layout>
+  `,
+})
+export class NbOptionGroupTestComponent {
+  selectDisabled: boolean = false;
+  optionGroupDisabled: boolean = false;
+  optionDisabled: boolean = false;
+  showOption: boolean = true;
+  optionGroupTitle: string = '';
+
+  @ViewChild(NbSelectComponent) selectComponent: NbSelectComponent<number>;
+  @ViewChild(NbOptionGroupComponent) optionGroupComponent: NbOptionGroupComponent;
+  @ViewChild(NbOptionComponent) optionComponent: NbOptionComponent<number>;
+}
+
+describe('NbOptionGroupComponent', () => {
+  let fixture: ComponentFixture<NbOptionGroupTestComponent>;
+  let testComponent: NbOptionGroupTestComponent;
+  let selectComponent: NbSelectComponent<number>;
+  let optionGroupComponent: NbOptionGroupComponent;
+  let optionComponent: NbOptionComponent<number>;
+
+  beforeEach(fakeAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        RouterTestingModule.withRoutes([]),
+        NbThemeModule.forRoot(),
+        NbLayoutModule,
+        NbSelectModule,
+      ],
+      declarations: [ NbOptionGroupTestComponent ],
+    });
+
+    fixture = TestBed.createComponent(NbOptionGroupTestComponent);
+    testComponent = fixture.componentInstance;
+    fixture.detectChanges();
+    flush();
+
+    selectComponent = testComponent.selectComponent;
+    optionGroupComponent = testComponent.optionGroupComponent;
+    optionComponent = testComponent.optionComponent;
+  }));
+
+  it('should contain passed title', () => {
+    const title = 'random option group title';
+    selectComponent.show();
+    testComponent.optionGroupTitle = title;
+    fixture.detectChanges();
+
+    const groupTitle = fixture.debugElement.query(By.directive(NbOptionGroupComponent))
+      .query(By.css('.option-group-title'));
+
+    expect(groupTitle.nativeElement.textContent).toEqual(title);
+  });
+
+  it('should have disabled attribute if disabled', () => {
+    selectComponent.show();
+    testComponent.optionGroupDisabled = true;
+    fixture.detectChanges();
+
+    const optionGroup = fixture.debugElement.query(By.directive(NbOptionGroupComponent));
+    expect(optionGroup.attributes.disabled).toEqual('');
+  });
+
+  it('should remove disabled attribute if disabled set to false', () => {
+    selectComponent.show();
+    testComponent.optionGroupDisabled = true;
+    fixture.detectChanges();
+
+    testComponent.optionGroupDisabled = false;
+    fixture.detectChanges();
+
+    const optionGroup = fixture.debugElement.query(By.directive(NbOptionGroupComponent));
+    expect(optionGroup.attributes.disabled).toEqual(null);
+  });
+
+  it('should disable options if set disabled', () => {
```

Comment:
I think that way test name will be more readable 😄 
```suggestion
  it('should disable group options if group disabled', () => {
```